lib/improver/metadata/set_attributes.py

```diff
@@ -0,0 +1,144 @@
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""
+Module setting attributes for IMPROVER blended data, nowcasts and spot
+forecasts
+"""
+
+import re
+import warnings
+
+from improver.metadata.constants.attributes import (
+    STANDARD_GRID_TITLE_STRING,
+    UK_SPOT_TITLE_STRING,
+    GLOBAL_SPOT_TITLE_STRING,
+    DATASET_ATTRIBUTES)
+from improver.utilities.cube_metadata import amend_metadata
+
+
+def set_product_attributes(cube, product):
+    """
+    Set attributes on an output cube of type matching a key string in the
+    DATASET_ATTRIBUTES dictionary.
+
+    Args:
+        cube (iris.cube.Cube):
+            Cube containing product data
+        product (str):
+            String describing product type
+
+    Returns:
+        updated_cube (iris.cube.Cube):
+            Cube with updated attributes
+    """
+    try:
+        original_title = cube.attributes["title"]
+    except KeyError:
+        original_title = ""
+
+    try:
+        updated_cube = amend_metadata(
+            cube, attributes=DATASET_ATTRIBUTES[product])
+    except KeyError:
+        options = list(DATASET_ATTRIBUTES.keys())
+        raise ValueError(
+            "product '{}' not available (options: {})".format(
+                product, options))
+
+    if STANDARD_GRID_TITLE_STRING in original_title:
+        updated_cube.attributes["title"] += " on {}".format(
+            STANDARD_GRID_TITLE_STRING)
+
+    return updated_cube
+
+
+def _match_title(original_title):
+    """
+    Match a title string to a regular expression
+
+    Args:
+        original_title (str):
+            Value of "title" attribute on an input cube
+
+    Returns:
+        match (str or None):
+            Match to expected regular expression pattern
+    """
+    regex = re.compile(
+        '(?P<field>.*?)'  # description of field
+        '( on )'          # expected joining statement
+        '(?P<grid>.*?)')  # eg STANDARD_GRID_TITLE_STRING
+    return regex.match(original_title)
+
+
+def update_spot_title_attribute(cube):
+    """
+    Update the "title" attribute on a spot data cube IF there is an existing
+    attribute of the form "<description>_on_<grid>" or containing the UK
+    standard grid definition string.  Modifies cube in place.
+
+    Args:
+        cube (iris.cube.Cube):
+            Spot data cube
+    """
+    try:
+        original_title = cube.attributes["title"]
+    except KeyError:
+        # if there is no title on the input cube, we cannot update it sensibly
+        return
+
+    if (UK_SPOT_TITLE_STRING in original_title or
+            GLOBAL_SPOT_TITLE_STRING in original_title):
+        # make no changes if title is already suitable
+        return
```

Comment:
Technically, there's no unit test covering this particular return statement.

Reply:
Added test.